frontend/src/utils/json-parser.ts

```diff
@@ -0,0 +1,166 @@
+        return null;
+      }
+      throw error;
+    }
+    const { reportGeneratedTime, reportGenerationTime } = data;
+    window.sinceDate = data.sinceDate;
+    window.untilDate = data.untilDate;
+    window.repoSenseVersion = data.repoSenseVersion;
+    window.isSinceDateProvided = data.isSinceDateProvided;
+    window.isUntilDateProvided = data.isUntilDateProvided;
+    document.title = data.reportTitle || document.title;
+
+    const errorMessages: { [key:string]: {errorMessage: string; repoName: string}; } = {};
+    Object.entries(data.errorSet).forEach(([repoName, message]) => {
+      errorMessages[repoName] = message;
+    });
+
+    window.DOMAIN_URL_MAP = data.supportedDomainUrlMap;
+
+    const names: string[] = [];
+    data.repos.forEach((repo) => {
+      const repoName = `${repo.displayName}`;
+      window.REPOS[repoName] = repo;
+      names.push(repoName);
+    });
+    return {
+      creationDate: reportGeneratedTime,
+      reportGenerationTime,
+      errorMessages,
+      names,
+    };
+  },
+
+  async loadCommits(repoName: string) {
+    const folderName = window.REPOS[repoName].outputFolderName;
+    const commits = await this.loadJSON(`${folderName}/commits.json`, 'commits');
+    const res: User[] = [];
+    const repo = window.REPOS[repoName];
+
+    Object.keys(commits.authorDisplayNameMap).forEach((author) => {
+      if (author) {
+        const obj: User = {
+          checkedFileTypeContribution: undefined,
+          commits: [],
+          dailyCommits: commits.authorDailyContributionsMap[author],
+          displayName: commits.authorDisplayNameMap[author],
+          fileTypeContribution: commits.authorFileTypeContributionMap[author],
+          location: '',
+          name: author,
+          repoId: repoName,
+          repoName: '',
+          searchPath: '',
+          variance: commits.authorContributionVariance[author],
+        };
+
+        this.setContributionOfCommitResultsAndInsertRepoId(obj.dailyCommits, obj.repoId);
+
+        const searchParams = [
+            repo.displayName,
+            obj.displayName, author,
+        ];
+
+        obj.searchPath = searchParams.join('_').toLowerCase();
+        obj.repoName = `${repo.displayName}`;
+        obj.location = `${repo.location.location}`;
+
+        res.push(obj);
+      }
+    });
+
+    repo.commits = commits;
+    repo.users = res;
+    return res;
+  },
+
+  loadAuthorship(repoName: string) {
+    const folderName = window.REPOS[repoName].outputFolderName;
+    return this.loadJSON(`${folderName}/authorship.json`, 'authorship')
+        .then((files: AuthorshipSchema) => {
+          window.REPOS[repoName].files = files;
+          return files;
+        });
+  },
+
+  // calculate and set the contribution of each commitResult and insert repoId into commitResult, since not provided in json file
+  setContributionOfCommitResultsAndInsertRepoId(dailyCommits: { date: string; commitResults: CommitResult[] }[], repoId: string) {
```

Comment:
I would still prefer that the dailyCommits parameter have its own type, maybe `dailyCommitType` or something